Formula/joern.rb

```diff
@@ -0,0 +1,73 @@
+class Joern < Formula
+  desc "Open-source code analysis platform based on code property graphs"
+  homepage "https://joern.io/"
+  url "https://github.com/joernio/joern/archive/refs/tags/v1.1.1384.tar.gz"
+  sha256 "c71b77c519b051a07728e20eb78538f6c76287e7ec2ee4ee082a8fd852f67dad"
+  license "Apache-2.0"
+
+  depends_on "node@18" => :build
+  depends_on "sbt" => :build
+  depends_on "yarn" => :build
+  depends_on "coreutils"
+  depends_on "openjdk@11"
+
+  # Astgen is a dependency of joern, but it is not available as a brew formula
```

Comment:
Then it should be made into a formula

Reply:
Sure thing! Would you suggest that be implemented as (A) a subformula (i.e., nesting `class Astgen < Formula` inside `class Joern < Formula` - if that's even possible), (B) a separate formula (`astgen.rb`) in this branch, or (C) a separate formula in a separate branch specifically for that formula?

Reply:
It should be a completely separate pull request.

Reply:
It appears that Joern uses [max-leuthaeuser/astgen](https://github.com/max-leuthaeuser/astgen) which is a modified fork of [joern/astgen](https://github.com/joernio/astgen). Attempting to create a Formula for this results in the following error on audit:

```bash
$ brew audit --new-formula astgen
astgen:
  * GitHub fork (not canonical repository)
Error: 1 problem in 1 formula detected
```

Do you have any advice on how to proceed?

Reply:
If they don't intend to ever contribute upstream, it should just be deforked.

Reply:
joern contributor here. We're looking into it. Thank you all for this effort!

Reply:
@mpollmeier if you're considering unforking, there is an semi-automated process described [here](https://stackoverflow.com/a/72550076). Another option would be to build `astgen` from source as part of the Joern build process. I know how to do this with `CMake`, but I'm unable to comment on how it would be done with `sbt` which Joern is using.

Reply:
My astgen fork lives now in joernio and will be de-forked soon.

Reply:
it's deforked: https://github.com/joernio/astgen
![grafik](https://user-images.githubusercontent.com/506752/212835013-5f3d5925-6c79-4cff-a1f4-94169ce9e2f9.png)

Reply:
@max-leuthaeuser @mpollmeier thank you both for migrating and deforking `astgen`.

@SMillerDev I attempted to create an Astgen formula for `Homebrew/homebrew-core`, but could not get the formula to pass the new formula audit due to failing the notability requirements:

```bash
$ brew audit --new-formula astgen
astgen:
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
Error: 1 problem in 1 formula detected
```

In an attempt to resolve this, I created the Astgen formula in a [Homebrew Tap](https://github.com/iandinwoodie/homebrew-tap/blob/main/Formula/astgen.rb) and locally adjusted the Joern to see if this would solve this audit issue. However, this solution also fails the new formula audit because dependencies on external taps are not allowed:

```bash
$ brew audit --new-formula joern
joern:
  * Dependency 'iandinwoodie/tap/astgen' is not in homebrew/core. Formulae in homebrew/core
    should not have dependencies in external taps.
```

Would you be amenable to allowing Astgen as a resource until it reaches the notability requirements; at which point I can submit my formula to `Homebrew/homebrew-core`?

Reply:
The newest release of astgen includes macOS ARM binaries, so I will eliminate this dependency/resource from the formula once https://github.com/joernio/joern/pull/2191 is integrated.

Reply:
👍

Reply:
Fixed as of https://github.com/Homebrew/homebrew-core/commit/b637d02b06d0ff043121d1b0ad51a91fc343de41